first_hour_on_vwb/working_with_data.ipynb

```diff
@@ -0,0 +1,589 @@
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "### Using the BigQuery Python client\n",
+    "\n",
+    "The cell below implements a widget using the [BigQuery Python client](https://github.com/googleapis/python-bigquery/tree/mainhttps://github.com/googleapis/python-bigquery/tree/main). To use the widget:\n",
+    "\n",
+    "1. Run the cell below to create the widget.\n",
+    "1. Select a table ID from the widget's dropdown of all the BigQuery tables available in your current workspace as resources.\n",
+    "1. Click the 'Resolve resource' button to view information about the table and produce an input field for a query.\n",
+    "1. Enter a query in the widget's text area.\n",
+    "1. Click 'Run query' to run the query and output the result in the widget.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "class ResolveDatasetWidget(object):\n",
+    "    def __init__(self):\n",
+    "        self.table = None\n",
+    "        self.client = bigquery.Client()\n",
+    "        self.label = widgets.Label(\n",
+    "            value=\"Please provide appropriate values in the input boxes.\")\n",
+    "        self.resource_names = self.get_bq_resources()\n",
+    "        self.name_dropdown = wu.DropdownInputWidget(\n",
+    "            self.resource_names, self.resource_names[0], \"Resource Name:\").get()\n",
+    "        self.input_query_textarea = widgets.Textarea(\n",
+    "            value='',\n",
+    "            placeholder='<INSERT QUERY>',\n",
+    "            description='Query:',\n",
+    "            layout=wu.input_layout,\n",
+    "            style=wu.input_style\n",
+    "        )\n",
+    "        self.query = None\n",
+    "        self.query_result = None\n",
+    "        self.output = widgets.Output()\n",
+    "        self.resolve_button = wu.StyledButton(\n",
+    "            \"Resolve resource\", \"Click to resolve a referenced resource\", \"check\",).get()\n",
+    "        self.query_button = wu.StyledButton(\n",
+    "            \"Query resource\", \"Click to query your referenced resource\", \"check\",).get()\n",
+    "        self.resolve_button.on_click(self.resolve_resource)\n",
+    "        self.query_button.on_click(self.run_query)\n",
+    "        self.vb = widgets.VBox(\n",
+    "            children=[\n",
+    "                self.label,\n",
+    "                self.name_dropdown,\n",
+    "                self.resolve_button,\n",
+    "                self.output\n",
+    "            ],\n",
+    "            layout=wu.vbox_layout\n",
+    "        )\n",
+    "\n",
+    "    def get_bq_resources(self):\n",
+    "        result = subprocess.run(\n",
+    "            [\"terra\", \"resource\", \"list\", \"--format=JSON\"], capture_output=True, text=True)\n",
+    "        ids_list = wu.list_bq_tables(result.stdout)\n",
```

Comment:
I'm not seeing the list_bq_tables function. Did you still need to check in those changes?

Reply:
fixed!